roles/common/tasks/check_upgrades.yml

```diff
@@ -20,54 +20,59 @@
         installed_components.acs is defined and
         installed_components.amps is defined and
         installed_components.api_explorer is defined and
-        ( 
+        (
           acs.edition != installed_components.acs.edition or
           acs.version.split('.')[:3] | join('.') != installed_components.acs.version.split('.')[:3] | join('.') or
           acs.version is version(installed_components.acs.version, 'lt') or
-          amps != installed_components.amps or
-          api_explorer != installed_components.api_explorer
+          api_explorer.version != installed_components.api_explorer.version
         )
 
+    - name: Fail on unsupported AMPS upgrades
+      fail:
+      when: |
+          installed_components.amps is defined and
+          amps[item].version != installed_components.amps[item].version
```

Comment:
you probably need something like `item in amps` otherwise you get an error if an amp is present in the system but not in the current vars?

try to reproduce by removing an amp from vars and then converge again

Reply:
I indeed reproduced something similar but after few adjustments to the checks I did it should no longer be a problem I hope